### PR TITLE
refactor: drop enum for `TuiCurrency`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/addon-commerce/enums/currency.ts [hunk 1]
```diff
@@ -397,5 +395,7 @@ export enum TuiCurrency {
      * @description Countries and territories: China
      * @description Currency symbol: CN¥
      */
-    YuanRenminbi = 'CNY',
-}
+    YuanRenminbi: 'CNY',
+} as const;
+
+export type TuiCurrencyT = (typeof TuiCurrency)[keyof typeof TuiCurrency];
```
Comment (waterplea) [inline projects/addon-commerce/enums/currency.ts:401]: We only used `xxxT` naming to distinguish between enum and joint type while we have both for backwards compatibility. Surely we shouldn't keep this name in the next major release.

Comment (splincode) [reply projects/addon-commerce/enums/currency.ts:401]: `TuiCurrencyType` is okey?

Comment (splincode) [reply projects/addon-commerce/enums/currency.ts]: fixed

**Review by waterplea — APPROVED** [review]
We should keep in mind migration of the type

**Pull request closed without merging**